src/cpp/src/tokenizer/chat_template_fallback_map.hpp

```diff
@@ -34,6 +34,10 @@ const std::pair<std::string, std::string> chat_template_fallback_map[] = {{
    "{% if messages is string %}{{ messages }}{% else %}{% for content in messages %}{% if content['type'] == 'image' or 'image' in content or 'image_url' in content %}<|vision_start|><|image_pad|><|vision_end|>{% elif content['type'] == 'video' or 'video' in content %}<|vision_start|><|video_pad|><|vision_end|>{% elif 'text' in content %}{{ content['text'] }}{% endif %}{% endfor %}{% endif %}",
     "{% for message in messages %}{{ message['content'] }}{% endfor %}"
 }, {
+   // gemma3-4b-it (tokenizer_config.json)
+    "{{ bos_token }}\n{%- if messages[0]['role'] == 'system' -%}\n    {%- if messages[0]['content'] is string -%}\n        {%- set first_user_prefix = messages[0]['content'] + '\n\n' -%}\n    {%- else -%}\n        {%- set first_user_prefix = messages[0]['content'][0]['text'] + '\n\n' -%}\n    {%- endif -%}\n    {%- set loop_messages = messages[1:] -%}\n{%- else -%}\n    {%- set first_user_prefix = \"\" -%}\n    {%- set loop_messages = messages -%}\n{%- endif -%}\n{%- for message in loop_messages -%}\n    {%- if (message['role'] == 'user') != (loop.index0 % 2 == 0) -%}\n        {{ raise_exception(\"Conversation roles must alternate user/assistant/user/assistant/...\") }}\n    {%- endif -%}\n    {%- if (message['role'] == 'assistant') -%}\n        {%- set role = \"model\" -%}\n    {%- else -%}\n        {%- set role = message['role'] -%}\n    {%- endif -%}\n    {{ '<start_of_turn>' + role + '\n' + (first_user_prefix if loop.first else \"\") }}\n    {%- if message['content'] is string -%}\n        {{ message['content'] | trim }}\n    {%- elif message['content'] is iterable -%}\n        {%- for item in message['content'] -%}\n            {%- if item['type'] == 'image' -%}\n                {{ '<start_of_image>' }}\n            {%- elif item['type'] == 'text' -%}\n                {{ item['text'] | trim }}\n            {%- endif -%}\n        {%- endfor -%}\n    {%- else -%}\n        {{ raise_exception(\"Invalid content type\") }}\n    {%- endif -%}\n    {{ '<end_of_turn>\n' }}\n{%- endfor -%}\n{%- if add_generation_prompt -%}\n    {{'<start_of_turn>model\n'}}\n{%- endif -%}\n",
+    "{{ bos_token }}<start_of_turn>{{ 'model' if messages[0]['role'] == 'assistant' else messages[0]['role'] }}\n{% if messages[0]['content'] is string %}{{ messages[0]['content'] | trim }}{% elif messages[0]['content'] is iterable %}{% for item in messages[0]['content'] %}{% if item['type'] == 'image' %}<start_of_image>{% elif item['type'] == 'text' %}{{ item['text'] | trim }}{% endif %}{% endfor %}{% endif %}<end_of_turn>\n{% if add_generation_prompt %}<start_of_turn>model{% endif %}",
```

Comment:
Chat template fallback for gemma3 is not totally correct and should be aligned/updated:
- trim filter affects prompt formatting
- in GenAI `message['content']` is always string - no need to check if it is iterable
- current chat template does not support system message for chat mode (`{ "role": "system", "content": ... }`)